tests/python/gaia-ui-tests/gaiatest/tests/functional/browser/test_browser_bookmark.py

```diff
@@ -36,17 +38,21 @@ def test_browser_bookmark(self):
 
         # Switch to Home Screen to look for bookmark
         self.device.touch_home_button()
-
         homescreen = Homescreen(self.marionette)
         homescreen.wait_for_app_icon_present(self.bookmark_title)
         self._bookmark_added = homescreen.is_app_installed(self.bookmark_title)
-
         self.assertTrue(self._bookmark_added, 'The bookmark %s was not found to be installed on the home screen.' % self.bookmark_title)
 
+        # verify that the bookmark goes to the correct URL
+        homescreen.installed_app(self.bookmark_title).tap_icon()
+        self.assertTrue(self.test_url == browser.apps.displayed_app.src, "The bookmark URL is the same as the browser URL")
+        self.device.touch_home_button()
+
+        homescreen = Homescreen(self.marionette)
         # Delete the bookmark
         homescreen.activate_edit_mode()
+        homescreen.bookmark(self.bookmark_title).tap_delete_app()
```

Comment:
The reason is because one is to tap the button, the other is to confirm.  I ran into a race condition where things would fail if I didn't tap the button twice.